src/main/java/org/opensha/commons/data/function/EvenlyDiscretizedFunc.java

```diff
@@ -339,6 +341,29 @@ public void set(double x, double y) {
 		points[index] = y;
 	}
 
+	/**
+	 * Maps a user-defined function F to each point, setting y = F(x).
+	 * 
+	 * @param mappedFn(x)
+	 */
+	public void setYofX(DoubleUnaryOperator mappedFn){
```

Comment:
You could easily put this as high in the hierarchy as `AbstractXY_DataSet`

Reply:
@voj Ahh, yes of course. For now I might leave it here and see what Kevins thoughts are about that.

Reply:
I like @voj's suggestion here, and may even end up using this stuff. I say add it to the XY_DataSet interface as a default implementation instead (only putting it in the abstract class means it won't work on plain `DiscretizedFunc` references, which are common).